drivers/counter/Kconfig.nrfx

```diff
@@ -3,108 +3,39 @@
-	bool "Counter on TIMER3"
-	depends on HAS_HW_NRF_TIMER3
-	depends on !NRF_HW_TIMER3_RESERVED
-	select COUNTER_NRF_TIMER
-
-config COUNTER_TIMER3_ZLI
-	bool "Event in ZLI interrupt context"
-	depends on COUNTER_TIMER3
-	depends on ZERO_LATENCY_IRQS
-
-config COUNTER_TIMER4
-	bool "Counter on TIMER4"
-	depends on HAS_HW_NRF_TIMER4
-	depends on !NRF_HW_TIMER4_RESERVED
-	select COUNTER_NRF_TIMER
-
-config COUNTER_TIMER4_ZLI
-	bool "Event in ZLI interrupt context"
-	depends on COUNTER_TIMER4
-	depends on ZERO_LATENCY_IRQS
-
-config COUNTER_RTC0
-	bool "Counter on RTC0"
-	depends on HAS_HW_NRF_RTC0
-	depends on !NRF_HW_RTC0_RESERVED
-	select COUNTER_NRF_RTC
-
-config COUNTER_RTC0_ZLI
-	bool "Event in ZLI interrupt context"
-	depends on COUNTER_RTC0
-	depends on ZERO_LATENCY_IRQS
-
-config COUNTER_RTC1
-	bool "Counter on RTC1"
-	depends on HAS_HW_NRF_RTC1
-	depends on !NRF_HW_RTC1_RESERVED
-	select COUNTER_NRF_RTC
-
-config COUNTER_RTC1_ZLI
-	bool "Event in ZLI interrupt context"
-	depends on COUNTER_RTC1
-	depends on ZERO_LATENCY_IRQS
-
-config COUNTER_RTC2
-	bool "Counter on RTC2"
-	depends on HAS_HW_NRF_RTC2
-	depends on !NRF_HW_RTC2_RESERVED
-	select COUNTER_NRF_RTC
-
-config COUNTER_RTC2_ZLI
-	bool "Event in ZLI interrupt context"
-	depends on COUNTER_RTC2
-	depends on ZERO_LATENCY_IRQS
+	default y
+	depends on DT_HAS_NORDIC_NRF_RTC_ENABLED
 
 # Internal flag which detects if PPI wrap feature is enabled for any instance
 config COUNTER_RTC_WITH_PPI_WRAP
-	def_bool ($(dt_nodelabel_bool_prop,rtc0,ppi-wrap) && COUNTER_RTC0) || \
-		 ($(dt_nodelabel_bool_prop,rtc1,ppi-wrap) && COUNTER_RTC1) || \
-		 ($(dt_nodelabel_bool_prop,rtc2,ppi-wrap) && COUNTER_RTC2)
+	def_bool $(dt_nodelabel_bool_prop,rtc0,ppi-wrap) || \
+		 $(dt_nodelabel_bool_prop,rtc1,ppi-wrap) || \
+		 $(dt_nodelabel_bool_prop,rtc2,ppi-wrap)
+	depends on COUNTER_NRF_RTC
 	select NRFX_PPI if HAS_HW_NRF_PPI
 	select NRFX_DPPI if HAS_HW_NRF_DPPIC
 
 # Internal flag which detects if fixed top feature is enabled for any instance
 config COUNTER_RTC_CUSTOM_TOP_SUPPORT
-	def_bool (!$(dt_nodelabel_bool_prop,rtc0,fixed-top) && COUNTER_RTC0) || \
-		 (!$(dt_nodelabel_bool_prop,rtc1,fixed-top) && COUNTER_RTC1) || \
-		 (!$(dt_nodelabel_bool_prop,rtc2,fixed-top) && COUNTER_RTC2)
+	def_bool !$(dt_nodelabel_bool_prop,rtc0,fixed-top) || \
+		 !$(dt_nodelabel_bool_prop,rtc1,fixed-top) || \
+		 !$(dt_nodelabel_bool_prop,rtc2,fixed-top)
+	depends on COUNTER_NRF_RTC
+
+# Internal flag which enables Zero Latency interrupts if any instance is using it
+config COUNTER_NRF_ZLI
+	def_bool $(dt_nodelabel_bool_prop,rtc0,zli) || \
+		 $(dt_nodelabel_bool_prop,rtc1,zli) || \
+		 $(dt_nodelabel_bool_prop,rtc2,zli) || \
+		 $(dt_nodelabel_bool_prop,timer0,zli) || \
+		 $(dt_nodelabel_bool_prop,timer1,zli) || \
+		 $(dt_nodelabel_bool_prop,timer2,zli) || \
+		 $(dt_nodelabel_bool_prop,timer3,zli) || \
+		 $(dt_nodelabel_bool_prop,timer4,zli)
+	depends on COUNTER_NRF_RTC || COUNTER_NRF_TIMER
+	select ZERO_LATENCY_IRQS
```

Comment:
I don't think this `select` is a good idea. Why not ensure instead with a build assertion in the driver that `ZERO_LATENCY_IRQS` is enabled when `zli` property is set for a given (enabled) instance?

Reply:
Removed. If ZLI is not enabled but used there is already a BUILD_ASSERT.